### PR TITLE
Workspace creation issue (UI-76)

### PR DESCRIPTION
**Pull request by Tanvir-rahman**
## What is the Purpose?
User don't see workspace creation option on first login

## What was the approach?
Solved param passing issue, default value and array check

## Are there any concerns to addressed further before or after merging this PR?
1) Was getting some warning in console. Probably it is occurring from navbar as we are dynamically generating router-link and not passing params. However didn't get any error. Any comment why this is happening exactly?
2) Dashboard is showing blank page. Is it normal behaviour?
Screenshot 2020-05-24 at 8 26 33 PM

## Mentions?
@michaelbukachi @andrewtpham 

## Issue(s) affected?
#76

### CODE REVIEW

**Comment by michaelbukachi** [comment]
@Tanvir-rahman Yes. The dashboard should be blank currently since those designs are still being worked on.

File: src/store/modules/workspace.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { getWorkspaces } from '@/api/workspace'
 
 const state = {
   workspaces: [],
-  defaultWorkspace: undefined,
+  defaultWorkspace: {},
```
Comment (michaelbukachi) [inline src/store/modules/workspace.js:5]: Why did you change this to `{}` ?

File: src/router/index.js [hunk 2]
```diff
@@ -60,7 +60,7 @@ const routes = [
   {
     path: '/',
     component: Main,
-    redirect: 'ws/redirect',
+    redirect: { name: 'WorkspaceRedirect' },
```
Comment (michaelbukachi) [inline src/router/index.js:63]: There's no advantage to using a name object instead of direct path since we are still in the `router` file.

File: src/store/modules/workspace.js (hunk 1 above)
Comment (Tanvir-rahman) [reply src/store/modules/workspace.js:5]: I always set variables to instead of defining it to null/undefined and then check if its empty or not (Like previous line - empty array). Was there any reason for assigning undefined?

File: src/router/index.js (hunk 2 above)
Comment (Tanvir-rahman) [reply src/router/index.js:63]: Otherwise if you change the route later, you have to change it from all places + someone can easily misspelled it. In named route you don't have to do that. So named route is better option i think.
May be we can set these standards project wise in other places too if there is not any specific reason.

Adding @ninetteadhikari  @andrewtpham @lupyana for your opinions too. And yours too @michaelbukachi

File: src/store/modules/workspace.js (hunk 1 above)
Comment (michaelbukachi) [reply src/store/modules/workspace.js:5]: An empty object is not a `falsy` value so it will always evaluate to `true` even it's empty. 
Here's the list of falsy values:
https://developer.mozilla.org/en-US/docs/Glossary/Falsy

File: src/router/index.js (hunk 2 above)
Comment (michaelbukachi) [reply src/router/index.js:63]: if you can misspell it in `path` you can misspell the `name` value also, don't you think so?:smile:

Comment (Tanvir-rahman) [reply src/router/index.js:63]: Yeah. That's possible.
Also we don't usually change name. And path can be very large over time. Name is shorter syntax. That's the practice i prefer. 😄

Comment (michaelbukachi) [reply src/router/index.js:63]: Haha. This is true. However, in this case 'path' shorter. Anyway, personal preferences of this kind need to be discussed beforehand. Not in the actual PR, especially if it's really debatable. Can you please change the code in the comment for `{}` so that I can merge this PR?

File: src/store/modules/workspace.js (hunk 1 above)
Comment (Tanvir-rahman) [reply src/store/modules/workspace.js:5]: Yeah. empty object is not falsy. 
Basically we are checking 
`const { slug } = this.$store.getters['workspace/getDefaultWorkspace']` in (redirect.vue file) based on property. Not on 'undefined' or empty object.

Comment (michaelbukachi) [reply src/store/modules/workspace.js:5]: Oh! I  see. I was unpacking the property. It's fine then. Let me merge.

File: src/router/index.js (hunk 2 above)
Comment (Tanvir-rahman) [reply src/router/index.js:63]: When i write something and give pr i always think about not 'in this case' but about future.
And by this PR we raised the Issue we might face. Optimisation and better practice should be followed from very much beginning. right? 
i added a comment for `{}` 🙂

Comment (michaelbukachi) [reply src/router/index.js:63]: Better practice is mostly based on personal opinion. Plus there are specific use cases you have to consider for instance in this one where the redirect is directly above the path so the changes of making a mistake are close to zero. It's rarely black or white. There's a lot of grey to consider :smiley:
